Mathlib/Tactic/Bound.lean

```diff
@@ -0,0 +1,280 @@
+/-
+Copyright (c) 2024 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+
+import Aesop
+import Mathlib.Algebra.Order.AbsoluteValue
+import Mathlib.Algebra.Order.BigOperators.Group.Finset
+import Mathlib.Algebra.Order.Field.Basic
+import Mathlib.Algebra.Order.Floor
+import Mathlib.Algebra.Order.Ring.Basic
+import Mathlib.Algebra.Order.Ring.Defs
```

Comment:
I'm always worried about importing significant theory in tactics files. It really ties our hands simplifying the import tree.

Could you either try to minimize these imports, or add some comments explaining why these are the minimum necessary?

Reply:
Yep, I will minimize this further.

Reply:
Done, the imports are now:

```
-- Bound/Init.lean
import Aesop.Frontend.Command

-- Bound/Attribute.lean
import Mathlib.Init.ZeroOne
import Mathlib.Tactic.Bound.Init
import Qq

-- Bound.lean
import Aesop
import Mathlib.Tactic.Bound.Attribute
import Mathlib.Tactic.Lemma
import Mathlib.Tactic.Linarith.Frontend
import Mathlib.Tactic.NormNum.Core
```

Reply:
If I get a pointer to how to use `match_expr` I can replace the `Qq` import with that.